src/main/java/roomescape/reservation/repository/JdbcReservationRepository.java

```diff
@@ -0,0 +1,132 @@
+package roomescape.reservation.repository;
+
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Repository;
+import roomescape.reservation.entity.Reservation;
+import roomescape.time.entity.ReservationTime;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class JdbcReservationRepository implements ReservationRepository {
+    private final RowMapper<Reservation> ROW_MAPPER = (resultSet, rowNum) -> {
+        final long id = resultSet.getLong("id");
+        String name = resultSet.getString("name");
+        LocalDate date = resultSet.getObject("date", LocalDate.class);
+        final long timeId = resultSet.getLong("time_id");
+        LocalTime timeValue = resultSet.getObject("start_at", LocalTime.class);
+        ReservationTime timeEntity = ReservationTime.of(timeId, timeValue);
+        final long themeId = resultSet.getLong("theme_id");
+        return Reservation.of(
+                id,
+                name,
+                date,
+                timeEntity,
+                themeId
+        );
```

Comment:
사소하지만, resultSet에서 결과를 꺼내올 때, `추출한 결과 == 주입할 값`이라면 별도로 변수 선언 없이 사용해도 괜찮지 않을까 싶어요. 😃 

```suggestion
        final long timeId = resultSet.getLong("time_id");
        LocalTime timeValue = resultSet.getObject("start_at", LocalTime.class);
        ReservationTime timeEntity = ReservationTime.of(timeId, timeValue);
        
        return Reservation.of(
            resultSet.getLong("id"),
            resultSet.getString("name"),
            resultSet.getObject("date", LocalDate.class),
            timeEntity,
            resultSet.getLong("theme_id")
        );
```